runtime/vam/expr/agg/mathfunc.go

```diff
@@ -192,3 +196,39 @@ func constToNumeric[T numeric](vec *vector.Const) T {
 		return T(val.Float())
 	}
 }
+
+func minString(state string, vec vector.Any) string {
+	switch vec := vec.(type) {
+	case *vector.Const:
+		if v, _ := vec.AsString(); v < state {
+			return v
+		}
+		return state
+	default:
+		for i := range vec.Len() {
+			v, _ := vector.StringValue(vec, i)
+			if v < state {
+				state = v
+			}
+		}
+		return state
+	}
```

Comment:
Nits: Indent less. And use the min built-in.

```suggestion
	if vec, ok  := vec.(*vector.Const); ok {
		return min(state, vec.Value().Ptr().AsString()
	}
	for i := range vec.Len() {
		v, _ := vector.StringValue(vec, i)
		state = min(state, v)
	}
	return state
```